inbm/dispatcher-agent/dispatcher/source/constants.py

```diff
@@ -25,16 +26,16 @@ class SourceParameters:
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationAddSourceParameters:
-    gpg_key_uri: str
-    gpg_key_name: str
     file_name: str
     sources: list[str] = field(default_factory=lambda: [])
+    gpg_key_uri: Optional[str] = field(default=None)
+    gpg_key_name: Optional[str] = field(default=None)
```

Comment:
**Suggestion:** Ensure that 'sources' list is validated to contain valid APT source entries to prevent potential issues with malformed sources. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    @dataclass(kw_only=True, frozen=True)
    class ApplicationAddSourceParameters:
        file_name: str
        sources: list[str] = field(default_factory=list)
        gpg_key_uri: Optional[str] = field(default=None)
        gpg_key_name: Optional[str] = field(default=None)

        def __post_init__(self):
            for source in self.sources:
                validate_apt_source(source)
```